client/packages/programs/src/JsonForms/components/BMI.tsx

```diff
@@ -6,26 +6,87 @@ import {
   uiTypeIs,
 } from '@jsonforms/core';
 import { withJsonFormsControlProps } from '@jsonforms/react';
-import { DetailInputWithLabelRow, NumUtils } from '@openmsupply-client/common';
-import { FORM_LABEL_WIDTH } from '../common';
+import {
+  BasicTextInput,
+  Box,
+  DetailInputWithLabelRow,
+  NumUtils,
+  ObjUtils,
+  Typography,
+  useTranslation,
+} from '@openmsupply-client/common';
+import { FORM_LABEL_WIDTH, JsonData, useZodOptionsValidation } from '../common';
+import { z } from 'zod';
+import { useEncounter, useProgramEvents } from '../../api';
 
 export const bmiTester = rankWith(10, uiTypeIs('BMI'));
 
 const round = (value: number) => Math.round(value * 100) / 100;
 
+const Options = z
+  .object({
+    /**
+     * Event type to check for "height" value. If not provided, it will always
+     * retrieve from "height" property of current form data.
+     */
+    eventType: z.string().optional(),
+  })
+  .strict();
+type Options = z.infer<typeof Options>;
+
+const usePreviousHeight = (
+  formData: JsonData | undefined,
+  eventType: string | undefined
+): { source: 'previous' | 'form' | undefined; height: number | undefined } => {
+  // fetch current encounter
+  const encounterId = useEncounter.utils.idFromUrl();
+  const { data: currentEncounter } = useEncounter.document.byId(encounterId);
+  const { data: events } = useProgramEvents.document.list(
+    {
+      filter: {
+        patientId: { equalTo: currentEncounter?.patient?.id ?? '' },
+        type: {
+          equalTo: eventType,
+        },
+        documentType: {
+          equalTo: currentEncounter?.type,
+        },
+      },
+      page: {
+        first: 1,
+      },
+    },
+    !!currentEncounter && !!eventType
```

Comment:
this enable flag says, if the query is enabled or not, you could only enable it if we need to fetch it e.g.

`const shouldFetchEvent = !formData?.['height'] && !!formData?.['weight'];`

and then:

 `!!currentEncounter && !!eventType && shouldFetchEvent`

Reply:
Yeah, thanks. fdc7f9efd

(Had to check that `formData` is not undefined too, but basically works)